### PR TITLE
boards: remove some of the non required variables export

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR cleans up some of the non required use of `export`. The variables are not used in external scripts but only by the build system itself (for configuration, etc), so there's no need to export them.

### Testing procedure

- A green Murdock
- **telosb and iotlab-m3**: verify that `make term` works (untested, I don't have this board...)
- **cc2650-launchpad** (and cc2650stk): verify that flashing a firmware works
- **arduino-due**: verify that flashing a firmware works with bossa
- **nz32-sc151**: verif that flashing with dfu works
- **esp32-wroom-32**: verify that `ENC28J60_PARAM_DEFINED` is only defined once (by adding `$(info DEBUG)` just before the variable is set 1 and checking the DEBUG message is only displayed once)
- **opencm904**: verify that flashing a firmware still works
- **stm32** (bluepill ?): verify that flashing a firmware with dfu (using blackmagic_dfu) still works

### Issues/PRs references

Related to #10850

### CODE REVIEW

File: boards/pic32-wifire/Makefile.include [hunk 1]
```diff
@@ -1,5 +1,5 @@
-export APPDEPS += $(RIOTCPU)/$(CPU)/$(CPU_MODEL)/$(CPU_MODEL).S
-export USE_UHI_SYSCALLS = 1
+APPDEPS += $(RIOTCPU)/$(CPU)/$(CPU_MODEL)/$(CPU_MODEL).S
+USE_UHI_SYSCALLS = 1
```
Comment (fjmolinas) [inline boards/pic32-wifire/Makefile.include]: This flag needs cleaning up, but not part of the PR.

File: boards/pic32-clicker/Makefile.include [hunk 2]
```diff
@@ -1,5 +1,5 @@
-export APPDEPS += $(RIOTCPU)/$(CPU)/$(CPU_MODEL)/$(CPU_MODEL).S
-export USE_UHI_SYSCALLS = 1
+APPDEPS += $(RIOTCPU)/$(CPU)/$(CPU_MODEL)/$(CPU_MODEL).S
```
Comment (fjmolinas) [inline boards/pic32-clicker/Makefile.include]: Exported in `vars`

File: boards/esp32-wroom-32/Makefile.dep [hunk 3]
```diff
@@ -7,7 +7,7 @@ ifneq (, $(filter enc28j60, $(USEMODULE)))
 
     # avoid multiple definitions when package depenedencies are resolved recursively
     ifndef ENC28J60_PARAM_DEFINED
-        export ENC28J60_PARAM_DEFINED = 1
```
Comment (fjmolinas) [inline boards/esp32-wroom-32/Makefile.dep]: Just to be sure @gschorcht there is not weird use case for why the export was needed?

File: boards/pic32-clicker/Makefile.include (hunk 2 above)
Comment (fjmolinas) [reply boards/pic32-clicker/Makefile.include]: can you add to the buildsystemcheck?

File: boards/esp32-wroom-32/Makefile.dep (hunk 3 above)
Comment (gschorcht) [reply boards/esp32-wroom-32/Makefile.dep]: Seems to work without the export. Thanks for cleaning up.

File: boards/pic32-clicker/Makefile.include (hunk 2 above)
Comment (aabadie) [reply boards/pic32-clicker/Makefile.include]: Done
